staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -126,17 +126,18 @@ func (a *Allocator) ClaimsToAllocate() []*resourceapi.ResourceClaim {
 // additional value. A name can also be useful because log messages do not
 // have a common prefix. V(5) is used for one-time log entries, V(6) for important
 // progress reports, and V(7) for detailed debug output.
-func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []resourceapi.AllocationResult, finalErr error) {
+func (a *Allocator) Allocate(ctx context.Context, node *v1.Node, extendedResourceClaim *resourceapi.ResourceClaim) (finalResult []resourceapi.AllocationResult, finalErr error) {
```

Comment:
As mentioned during the KEP review, I don't like adding a special case for `extendedResourceClaim`. It's not just the additional parameter here, but also the additional if checks below.

We don't gain anything from passing the set of claims to `NewAllocator`. Let's move that parameter here and into the `allocator` struct, then the caller is the only place where we need to deal with the special ResourceClaim.